homeassistant/components/hue/scene.py

```diff
@@ -57,10 +57,16 @@ def async_add_entity(event_type: EventType, resource: HueScene) -> None:
     platform.async_register_entity_service(
         SERVICE_ACTIVATE_SCENE,
         {
-            vol.Optional(ATTR_DYNAMIC): vol.Coerce(int),
-            vol.Optional(ATTR_SPEED): vol.Coerce(int),
-            vol.Optional(ATTR_TRANSITION): vol.Coerce(int),
-            vol.Optional(ATTR_BRIGHTNESS): vol.Coerce(int),
+            vol.Optional(ATTR_DYNAMIC): vol.Coerce(bool),
+            vol.Optional(ATTR_SPEED): vol.All(
+                vol.Coerce(int), vol.Clamp(min=0, max=100)
```

Comment:
I would suggest to use vol.Range so we raise an error if incorrect values are sent instead of silently fixing it .

Reply:
adjusted